scripts/kubestellar-snapshot.sh

```diff
@@ -0,0 +1,580 @@
+    (--verbose|-V)
+        arg_verbose=true;;
+    (--version|-v)
+        echo "${SCRIPT_NAME} v${SCRIPT_VERSION}"
+        exit 0;;
+    (-X)
+        set -x;;
+    (-h|--help)
+        display_help
+        exit 0;;
+    (-*)
+        echo "$0: unknown flag" >&2
+        exit 1;;
+    (*)
+        echo "$0: unknown positional argument" >&2
+        exit 1;;
+    esac
+    shift
+done
+
+
+###############################################################################
+# Alias definitions
+###############################################################################
+# Define the echov function based on verbosity
+if [ "$arg_verbose" == "true" ]; then
+    echov() { echo "$@" ; }
+else
+    echov() { :; }
+fi
+
+
+###############################################################################
+# Script info
+###############################################################################
+echov -e "${COLOR_INFO}${SCRIPT_NAME} v${SCRIPT_VERSION}{COLOR_NONE}\n"
+echov -e "Script run on ${COLOR_INFO}$TIMESTAMP${COLOR_NONE}"
+
+
+###############################################################################
+# Check dependencies
+###############################################################################
+echov "Checking script dependencies:"
+
+is_installed 'kubectl' \
+    'kubectl' \
+    'kubectl version --client | head -1 | cut -d" " -f3'
+
+is_installed 'helm' \
+    'helm' \
+    'helm version | cut -d"\"" -f2'
+
+is_installed 'jq' \
+    'jq' \
+    'jq --version'
+
+
+###############################################################################
+# Ensure output folder
+###############################################################################
+if [[ "$arg_logs" == "true" || "$arg_yaml" == "true" ]] ; then
+    mkdir -p "$OUTPUT_FOLDER"
+fi
+
+
+###############################################################################
+# Determine the list of kubeconfigs to search
+# based on https://kubernetes.io/docs/reference/k/generated/k_config/
+###############################################################################
+[[ -n "$arg_kubeconfig" ]] && export KUBECONFIG="$arg_kubeconfig"
+[[ -z "$KUBECONFIG" ]] && export KUBECONFIG="$HOME/.kube/config"
+echov -e "Using kubeconfig(s): ${COLOR_INFO}$KUBECONFIG${COLOR_NONE}"
+
+
+###############################################################################
+# Determine the list of contexts to search
+###############################################################################
+if [[ "$arg_context" == "" ]] ; then
+    contexts=($(kubectl config get-contexts --no-headers -o name))
+else
+    contexts=("$arg_context")
+fi
+
+if [[ -z "${contexts[@]}" ]] ; then
+    echoerr "No context(s) found in the kubeconfig file $KUBECONFIG!"
+    exit 1
+fi
+
+echov "Validating contexts(s): "
+valid_contexts=()
+for context in "${contexts[@]}" ; do # for all contexts
+    [[ -z "$context" ]] && continue
+    [[ "$context" == "$(kubectl config current-context || true)" ]] && cc="*" || cc=""
```

Comment:
FYI: `$(kubectl config current-context || true)` does not need to be evaluated on every iteration, it could be done once before the loop.